### ISSUE TITLE
SourceGenerator doesnt generate if in global namespace

### ISSUE THREAD

**Issue by OpenByteDev** [corvus-dotnet/Corvus.JsonSchema#906]
This doesnt generate any code:
```
using Corvus.Json;

[JsonSchemaTypeGenerator("schema.json")]
public readonly partial struct StringOrExpression;
```

While this works just fine:
```
using Corvus.Json;

namespace Test;

[JsonSchemaTypeGenerator("schema.json")]
public readonly partial struct StringOrExpression;
```

**Comment by mwadams** [comment]
## Root cause

The generator's syntactic pre-filter rejects global-namespace targets before the pipeline ever sees them.

`Initialize` wires `ForAttributeWithMetadataName` with the predicate `IsValidAttributeTarget` (`src/Corvus.Text.Json.SourceGenerator/IncrementalSourceGenerator.cs`), which requires:

```csharp
structDeclarationSyntax.Parent is (FileScopedNamespaceDeclarationSyntax or NamespaceDeclarationSyntax)
```

For a struct declared in the global namespace, the parent is a `CompilationUnitSyntax`, so the predicate returns false and the attributed struct never reaches `BuildGenerationSpecifications`. Nothing downstream runs and no diagnostic is reported, which is why the failure is silent. The V4 generator (`src-v4/Corvus.Json.SourceGenerator/IncrementalSourceGenerator.cs`) has the identical predicate, so this limitation was inherited, not introduced.

Reproduced with a minimal net10.0 project (attributed partial struct in the global namespace, schema in `AdditionalFiles`): the build succeeds, `EmitCompilerGeneratedFiles` shows only `JsonSchemaTypeGeneratorAttribute.g.cs`, and no types are generated.

There are two further layers behind the filter, confirmed empirically by relaxing it and rebuilding the repro:

1. `BuildGenerationSpecifications` computes the namespace as `context.TargetSymbol.ContainingNamespace.ToDisplayString()`, which yields the literal string `<global namespace>` for a global-namespace type. The four query-language source generators in this repo already handle this and provide the in-repo precedent (`IsGlobalNamespace ? string.Empty : ToDisplayString()`).
2. With the namespace mapped to the empty string, the C# emission produces exactly two classes of broken output, both mechanical:
   - `namespace ;` lines (from `BeginNamespace`, the global-declarations helper in `CSharpLanguageProvider`, and the standalone evaluator generator), and
   - leading-dot fully qualified names such as `.StringOrExpression.RequiredExpression` (from `FullyQualifiedDotnetTypeName` building `$"{ns}.{name}"` with an empty namespace).

Nothing else breaks. The member-scope machinery uses scope strings only as opaque keys, file names are already namespace-free (`DotnetTypeNameWithoutNamespace`), and no emission site prepends `global::` to these names.

## Plan (validated against the repro)

1. **Relax `IsValidAttributeTarget`** to also accept a struct whose parent is a `CompilationUnitSyntax`.
2. **Map the global namespace to `string.Empty`** in `BuildGenerationSpecifications`, mirroring the query-language generators.
3. **Handle the empty namespace at the four emission sites**:
   - `CodeGeneratorExtensions.BeginNamespace`: skip the `namespace ...;` line when the namespace is empty, but still push the member scope so Begin/End pairing holds;
   - `TypeDeclarationExtensions.FullyQualifiedDotnetTypeName`: no leading dot for root types with an empty namespace;
   - `CSharpLanguageProvider` global-declarations helper: same two adjustments for `DefaultNamespace`;
   - `StandaloneEvaluatorGenerator.EmitNamespaceOpen`: skip the namespace line when empty.
4. **Regression tests**: add a global-namespace attributed struct (`GlobalNamespaceModel`, `oneOf` string/object like the sample in this issue) to `tests/Corvus.Text.Json.Tests.GeneratedModels`, and tests in `tests/Corvus.Text.Json.Tests` that parse and schema-evaluate all three shapes. This proves generation end to end through the real build on every TFM the models project targets, and a regression to silent non-generation fails the build rather than passing quietly.
5. **Gates**: warning-free `Corvus.Text.Json.slnx` build; the standard filtered test run on all TFMs. The emission changes only alter behaviour when the namespace is empty, which no existing path produces, so no regeneration of existing generated code is needed.

With this in place the repro generates compiling code in the global namespace, and the `oneOf` behaves correctly at runtime (string form valid, object form valid, number rejected).

The fix lands on `feature/910-906-message-whitespace-and-global-namespace`, layered with the #910 fix.
